### PR TITLE
Consolidate CRD metadata handling

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/18019

The main goal of this PR is to simplify CRD metadata handling.

Before, metadata and CRD docs generation config was handled in two separate files, both here in this repo.

With https://github.com/giantswarm/apiextensions/pull/755, CRD metadata is moving into giantswarm/apiextensions.

The file `src/data/crd_metadata.yaml` is becoming obsolete, as the metadata from giantswarm/apiextensions is written into CRD page front matter via crd-docs-generator as of https://github.com/giantswarm/crd-docs-generator/pull/44

Some CRDs that were previously published are now marked hidden in apiextensions, hence the removal here.

## TODO

- [x] Merge https://github.com/giantswarm/apiextensions/pull/755 and release
- [x] Merge https://github.com/giantswarm/crd-docs-generator/pull/44 and release
- [x] Update apiextensions and crd-docs-generator versions in `scripts/update-crd-reference/main.sh`

### CODE REVIEW

File: scripts/update-crd-reference/config.yaml [hunk 1]
```diff
@@ -3,31 +3,5 @@ source_repository:
...
-  - vspheremachinetemplates.infrastructure.cluster.x-k8s.io
-  - vspherevms.infrastructure.cluster.x-k8s.io
-
-  # CAPI for AWS to be hidden until supported for customers
-  - awsclusterroleidentities.infrastructure.cluster.x-k8s.io
-  - awsclusters.infrastructure.cluster.x-k8s.io
-  - awsmachinepools.infrastructure.cluster.x-k8s.io
-  - awsmachinetemplates.infrastructure.cluster.x-k8s.io
-  - kubeadmconfigs.bootstrap.cluster.x-k8s.io
-  - kubeadmcontrolplanes.controlplane.cluster.x-k8s.io
```
Comment (marians) [inline scripts/update-crd-reference/config.yaml:33]: This is moving to https://github.com/giantswarm/apiextensions/pull/755/files

File: scripts/update-crd-reference/config.yaml [hunk 2]
```diff
@@ -3,31 +3,5 @@ source_repository:
...
+  metadata_path: docs/config.yaml
+template_path: crd.template
```
Comment (marians) [inline scripts/update-crd-reference/config.yaml:7]: This was a command line flag before

File: scripts/update-crd-reference/crd.template [hunk 3]
```diff
@@ -1,19 +1,44 @@
...
+    - {{ . -}}
+{{- end }}
+  topics:
+{{- range .Topics }}
+    - {{ . -}}
+{{- end }}
+  providers:
+{{- range .Providers }}
+    - {{ . -}}
+{{- end }}
```
Comment (marians) [inline scripts/update-crd-reference/crd.template:31]: This places all CRD related data into a common key `.crd`.

For compatibility reasons, existing front matter keys on the top level, like `technical_name`, `source_repository` and `source_repository_ref` are left intact. Once docs_indexer is adapted to the new schema, we can remove the duplication.

File: scripts/update-crd-reference/main.sh [hunk 4]
```diff
@@ -2,14 +2,18 @@
...
 
 # Clear output folder
 find ${DESTINATION} -type f -not -name "_index.md" | xargs rm
 
 # Determine apiextensions version
-VERSION=$(docker run --rm --entrypoint /bin/sh quay.io/giantswarm/docs-scriptrunner -c "curl --silent https://api.github.com/repos/giantswarm/apiextensions/releases/latest|jq -r .tag_name|tr -d '\n'")
+#VERSION=$(docker run --rm --entrypoint /bin/sh quay.io/giantswarm/docs-scriptrunner -c "curl --silent https://api.github.com/repos/giantswarm/apiextensions/releases/latest|jq -r .tag_name|tr -d '\n'")
+
+# TODO: revert this
+VERSION=add-docs-metadata
```
Comment (marians) [inline scripts/update-crd-reference/main.sh:16]: To be reverted

File: scripts/update-crd-reference/main.sh [hunk 5]
```diff
@@ -19,5 +23,4 @@ docker run --rm \
     -v ${PWD}/scripts/update-crd-reference:/opt/crd-docs-generator/config \
     quay.io/giantswarm/crd-docs-generator:${CRD_DOCS_GENERATOR_VERSION} \
         --commit-reference ${VERSION} \
-        --config /opt/crd-docs-generator/config/config.yaml \
-        --template /opt/crd-docs-generator/config/crd.template
```
Comment (marians) [inline scripts/update-crd-reference/main.sh:23]: `--template` flag moved into the config file

File: scripts/validate-front-matter/script.py [hunk 6]
```diff
@@ -179,6 +179,7 @@
     'aliases',
     'changes_categories',
     'changes_entry',
+    'crd',
```
Comment (marians) [inline scripts/validate-front-matter/script.py:182]: A new top level front matter key to accept

File: src/content/ui-api/management-api/crd/appcatalogentries.application.giantswarm.io.md [hunk 7]
```diff
@@ -1,15 +1,31 @@
 ---
-title: AppCatalogEntry CRD schema reference
+title: AppCatalogEntry CRD schema reference (group application.giantswarm.io)
```
Comment (marians) [inline src/content/ui-api/management-api/crd/appcatalogentries.application.giantswarm.io.md:2]: Adding the group name to the page title, to make CRD pages with identical short names easier to distinguish from search results.

File: scripts/update-crd-reference/crd.template [hunk 8]
```diff
@@ -1,19 +1,44 @@
 ---
-title: {{ .Title }} CRD schema reference
+title: {{ .Title }} CRD schema reference (group {{ .Group }})
```
Comment (marians) [inline scripts/update-crd-reference/crd.template:2]: Adding the group name to the page title, to make CRD pages with identical short names easier to distinguish from search results.

**Review by axbarsan — APPROVED** [review]
nice

**Pull request merged**